File: Codeforces/p01b_SpreadSheets/spreadSheet.cpp
#include<iostream>
#include<string>
#include<regex>

using namespace std;

// RXCY -> Alphanumeric
void rowCoordinate(string input){
    int cIndex = input.find('C');

    string row = input.substr(1, cIndex -1);
    string col = input.substr(cIndex + 1);

    long long num = stoll(col);
    string colString = "";

    while(num > 0){
        num--;
        long long remainder = num % 26;
        char ch = 'A' + remainder;
        colString = ch + colString;
        num /= 26;
    }
    
    cout << colString << row << "\n";
}

// Alphanumeric -> RXCY
void rowColSystem(string input){
    int firstDigitIndex = 0;

    while(firstDigitIndex < input.size() && !isdigit(input[firstDigitIndex])){
        firstDigitIndex++;
    }

    string row = input.substr(firstDigitIndex);
    string col = input.substr(0, firstDigitIndex);

    long long total = 0;

    for(char letter : col){
        int value = letter - 'A' + 1;
        total = total * 26 + value;
    }

    cout << "R" << row << "C" << total << "\n";
}

int main(){
    int n;
    cin >> n;

    regex pattern("^R[0-9] + C[0-9]+$");

    while(n--){
        string str;
        cin >> str;

        if(regex_match(str, pattern)){
            rowCoordinate(str);
        } else{
            rowColSystem(str);
        }
    }

    return 0;
}

File: Codeforces/p01b_SpreadSheets/spreadSheet.py
import re

# RXCY -> Alphanumeric
def row_coordinate(input_str):
    c_Index = input_str.index("C")

    row = input_str[1 : c_Index]
    col = input_str[c_Index + 1:]

    num = int(col)
    col_string = ""

    while num > 0:
        num -= 1
        remainder = num % 26
        col_string = chr(ord('A') + remainder) + col_string
        num //= 26
    
    print(col_string + row)

# Alphanumeric -> RXCY
def row_col_system(str):
    first_digit = 0
    while(first_digit < len(str) and not str[first_digit].isdigit()):
        first_digit += 1

    row = str[first_digit:]
    col = str[:first_digit]

    total = 0

    for letter in col:
        value = ord(letter) - ord('A') + 1
        total = total * 26 + value

    print(f"R{row}C{total}")

n = int(input())

for _ in range(n):
    s = input()

    if re.fullmatch(r"R/d+C/d+", s):
        row_coordinate(s)
    else:
        row_col_system(s)